events/exchange.go

```diff
@@ -161,3 +180,46 @@ func validateTopic(topic string) error {
 
 	return nil
 }
+
+func validateEnvelope(envelope *events.Envelope) error {
+	if err := namespaces.Validate(envelope.Namespace); err != nil {
+		return errors.Wrapf(err, "event envelope has invalid namespace")
+	}
+
+	if err := validateTopic(envelope.Topic); err != nil {
+		return errors.Wrapf(err, "envelope topic %q", envelope.Topic)
```

Comment:
The error message comes from the validate function. Adding more here is redundant.